app/controllers/tweetscontroller.rb

```diff
@@ -0,0 +1,85 @@
+class TweetsController < ApplicationController
+
+    
+
+    # get '/show' do
+    #     @tweets = Tweet.all
+    #     erb :'/show'
+    # end
+
+    get '/tweets/new' do
+        if logged_in?
+             erb :'tweets/new'
+        else # logged out and doesn't let them view the create tweet
+
+        redirect to "/login"
+        end
+    end
+
+     post '/tweets' do
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.